src/core/document.js

```diff
@@ -715,6 +697,61 @@ class PDFDocument {
     return shadow(this, "numPages", num);
   }
 
+  /**
+   * @private
+   */
+  _hasOnlyDocumentSignatures(fields) {
+    return fields.every(field => {
+      field = this.xref.fetchIfRef(field);
+      if (field.has("Kids")) {
+        return this._hasOnlyDocumentSignatures(field.get("Kids"));
+      }
+      const fieldType = field.get("FT");
+      const isSignature = isName(fieldType) && fieldType.name === "Sig";
```

Comment:
This can be actually be simplified to the e.g. following:
```suggestion
      const isSignature = isName(/* field type = */ field.get("FT"), "Sig");
```